partner_contact_depth/models/res_partner.py

```diff
@@ -0,0 +1,62 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def _get_partners_with_no_parent(self):
```

Comment:
does this code do the same as this other?

```suggestion
    def _get_partners_with_no_parent(self):
        return self.search([('parent_id', '=', False)])
```

Reply:
The aim it's just to recompute hierarchy related to partners in self, otherwise we will be recomputing depth for all partners.